e2e/testcases/override_reconcile_timeout_test.go

```diff
@@ -68,88 +78,55 @@ func TestOverrideReconcileTimeout(t *testing.T) {
 	nt.RootRepos[configsync.RootSyncName].Add("acme/ns-1.yaml", fake.NamespaceObject(namespaceName))
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Add namespace/%s & pod/%s (never ready)", namespaceName, pod1Name))
 	nt.WaitForRepoSyncs()
+	expectActuationStatus := "Succeeded"
+	expectReconcileStatus := "Timeout"
+	nomostest.WatchForObject(nt, kinds.ResourceGroup(), "root-sync", "config-management-system",
+		[]nomostest.Predicate{
+			resourceStatusEquals(idToVerify, expectActuationStatus, expectReconcileStatus),
+		})
 
-	u := &unstructured.Unstructured{}
-	u.SetGroupVersionKind(schema.GroupVersionKind{
-		Group:   "kpt.dev",
-		Kind:    "ResourceGroup",
-		Version: "v1alpha1",
-	})
-
-	err := nt.Client.Get(nt.Context, client.ObjectKey{
-		Namespace: "config-management-system",
-		Name:      "root-sync",
-	}, u)
-	if err != nil {
-		nt.T.Fatal(err)
-	} else {
-		expectActuationStatus := "Succeeded"
-		expectReconcileStatus := "Timeout"
-		err = verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus)
-		if err != nil {
-			nt.T.Fatal(err)
-		} else {
-			nt.T.Logf("object %q has actuation and reconcile status as expected", namespaceName)
-		}
-	}
+	// Override reconcileTimeout to 5m, namespace actuation should succeed, namespace reconcile should succeed.
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "5m"}}}`)
+	nt.WaitForRepoSyncs()
 
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/pod-1.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Remove pod/%s (never ready)", pod1Name))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Remove pod/%s", pod1Name))
 	nt.WaitForRepoSyncs()
 
 	// Verify pod is deleted.
 	// Otherwise deletion of a managed resource will be denied by the webhook.
 	nt.MustKubectl("delete", "pods", pod1Name, "-n", namespaceName, "--ignore-not-found")
```

Comment:
This seems like a confusing way to validate deletion, especially if the webhook isn't enabled by default.

Don't we have a ValidateNotFound or similar?

Reply:
Changed to ValidateNotFound